core/src/main/java/io/questdb/griffin/engine/functions/catalogue/FunctionListFunctionFactory.java

```diff
@@ -0,0 +1,283 @@
+            }
+
+            @Override
+            public boolean hasNext() {
+                if (funcNameIndex < funcNames.size() - 1) {
+                    if (funcDescriptors == null) {
+                        funcNameIndex++;
+                        CharSequence funcName = funcNames.get(funcNameIndex);
+                        if (isExcluded(funcName)) {
+                            return hasNext();
+                        }
+                        funcDescriptors = factories.get(funcName);
+                        descriptorIndex++;
+                        record.init(funcName, funcDescriptors.get(descriptorIndex).getFactory());
+                        return true;
+                    } else {
+                        descriptorIndex++;
+                        if (descriptorIndex < funcDescriptors.size()) {
+                            record.init(
+                                    funcNames.get(funcNameIndex),
+                                    funcDescriptors.get(descriptorIndex).getFactory()
+                            );
+                            return true;
+                        } else {
+                            descriptorIndex = -1;
+                            funcDescriptors = null;
+                            return hasNext();
+                        }
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public void recordAt(Record record, long atRowId) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public long size() {
+                return -1;
+            }
+
+            @Override
+            public void toTop() {
+                close();
+            }
+
+            private class FunctionRecord implements Record {
+                private FunctionFactory funcFactory;
+                private CharSequence funcName;
+
+                @Override
+                public boolean getBool(int col) {
+                    if (col == RT_CONSTANT_COLUMN) {
+                        return funcFactory.isRuntimeConstant();
+                    }
+                    throw new IllegalArgumentException("offending: " + col);
+                }
+
+                @Override
+                public CharSequence getStr(int col) {
+                    if (col == NAME_COLUMN) {
+                        return funcName;
+                    }
+                    if (col == SIGNATURE_COLUMN) {
+                        return funcFactory.getSignature();
+                    }
+                    if (col == SIGNATURE_TRANSLATED_COLUMN) {
+                        try {
+                            return FunctionFactoryDescriptor.translateSignature(funcFactory.getSignature());
+                        } catch (SqlException err) {
+                            // this cannot happen, all signatures are extracted programmatically
```

Comment:
Returning exception text as column value is weird. 
I think it'd better to return null and, optionally, log the error instead .

Reply:
addressed in a1a73ca10077447f80b5c96f5ce3d8553ceb85f5